lib/auth/auth.go

```diff
@@ -621,9 +621,48 @@ type Services struct {
 	services.DevicesGetter
 }
 
-// SecReportsClient returns the security reports client.
-func (r *Services) SecReportsClient() *secreport.Client {
-	return nil
+// NewAccessCacheForServices creates a new cache for a Teleport service that
+// uses the provided services to populate its resource collections.
+func NewAccessCacheForServices(cfg accesspoint.AccessCacheConfig, services *Services) (*cache.Cache, error) {
```

Comment:
I'd originally wanted this function to be in `accesspoint`, but unfortunately, this creates an import loop due to `NewTestAuthServer` in `helpers.go`.

Reply:
Refactoring `auth.Services` out of auth package would solve the problem and simplify the `auth` package itself

Reply:
[ad203d8](https://github.com/gravitational/teleport/pull/45058/commits/ad203d889881203b6eddc6e5c4f97fb13a5bf268)